pkg/util/node/node.go

```diff
@@ -158,14 +158,29 @@ func GetZoneKey(node *v1.Node) string {
 	return region + ":\x00:" + failureDomain
 }
 
+type nodeForPatch struct {
```

Comment:
the types are getting a bit confusing, naming the top level per use might be easier to follow

```suggestion
type nodeForConditionPatch struct {
```